tests/transports/test_base.py

```diff
@@ -240,3 +240,12 @@ def test_compress_level_sanitization():
     assert DummyTransport(compress_level=None, url="", client=None)._compress_level == 0
     assert DummyTransport(compress_level=-1, url="", client=None)._compress_level == 0
     assert DummyTransport(compress_level=10, url="", client=None)._compress_level == 9
+
+
+@mock.patch("elasticapm.transport.base.Transport.send")
+def test_transport_metadata_pid_change(mock_send, elasticapm_client):
+    transport = Transport(client=elasticapm_client, metadata={})
```

Comment:
how about putting something into the `metadata` dict and see if that shows up in `transport._metadata`?

Reply:
The only reason I left in the ability to provide overrides is that without that addition, any test that provided metadata would lose that metadata at `transport.start_thread()` time. So, for example, [this test](https://github.com/elastic/apm-agent-python/blob/a00659d3d49e89a1e3854f7663409feb79ae2426/tests/transports/test_base.py#L101) would lose the `{"x": "y"}`. 

Though, now that I'm looking through, I don't think any tests actually rely on those values. That test provides those values just to make sure some metadata is sent, but with the new change it definitely will be sent, as we'll generate the whole metadata dictionary at `start_thread()`. Based on that, I'll probably just remove the metadata kwarg altogether and see if anything breaks.

Reply:
Done.